apps/meteor/client/views/admin/settings/inputs/TimespanSettingInput.tsx

```diff
@@ -0,0 +1,136 @@
+import { Field, FieldLabel, FieldRow, InputBox, Select } from '@rocket.chat/fuselage';
+import { useTranslation } from '@rocket.chat/ui-contexts';
+import type { FormEventHandler, ReactElement } from 'react';
+import React, { useMemo, useState } from 'react';
+
+import ResetSettingButton from '../ResetSettingButton';
+import type { SettingInputProps } from './types';
+
+type TimespanSettingInputProps = SettingInputProps<string, string | number> & {
+	value: string;
+};
+
+export enum TIMEUNIT {
+	days = 'days',
+	hours = 'hours',
+	minutes = 'minutes',
+}
+
+export const timeUnitToMs = (unit: TIMEUNIT, timespan: number) => {
+	if (unit === TIMEUNIT.days) {
+		return timespan * 24 * 60 * 60 * 1000;
+	}
+
+	if (unit === TIMEUNIT.hours) {
+		return timespan * 60 * 60 * 1000;
+	}
+
+	if (unit === TIMEUNIT.minutes) {
+		return timespan * 60 * 1000;
+	}
+
+	throw new Error('TimespanSettingInput - timeUnitToMs - invalid time unit');
+};
+
+export const msToTimeUnit = (unit: TIMEUNIT, timespan: number) => {
+	if (unit === TIMEUNIT.days) {
+		return timespan / 24 / 60 / 60 / 1000;
+	}
+
+	if (unit === TIMEUNIT.hours) {
+		return timespan / 60 / 60 / 1000;
+	}
+
+	if (unit === TIMEUNIT.minutes) {
+		return timespan / 60 / 1000;
+	}
+
+	throw new Error('TimespanSettingInput - msToTimeUnit - invalid time unit');
+};
+
+export const getHighestTimeUnit = (value: number): TIMEUNIT => {
+	const minutes = msToTimeUnit(TIMEUNIT.minutes, value);
+	if (minutes % 60 !== 0) {
+		return TIMEUNIT.minutes;
+	}
+
+	const hours = msToTimeUnit(TIMEUNIT.hours, value);
+	if (hours % 24 !== 0) {
+		return TIMEUNIT.hours;
+	}
+
+	return TIMEUNIT.days;
+};
+
+function TimespanSettingInput({
+	_id,
+	label,
+	value,
+	placeholder,
+	readonly,
+	autocomplete,
+	disabled,
+	required,
+	onChangeValue,
+	hasResetButton,
+	onResetButtonClick,
+}: TimespanSettingInputProps): ReactElement {
+	const t = useTranslation();
+
+	const [timeUnit, setTimeUnit] = useState<TIMEUNIT>(getHighestTimeUnit(Number(value)));
+	const [internalValue, setInternalValue] = useState<number>(msToTimeUnit(timeUnit, Number(value)));
+
+	const handleChange: FormEventHandler<HTMLInputElement> = (event) => {
+		const newValue = Math.max(1, Number(event.currentTarget.value)) || 1;
+
+		try {
+			setInternalValue(newValue);
```

Comment:
I think we should validate the newvalue (in the server as well I guess), not sure how high it would have to be, but I guess if we put the value in days, and it is large enough we would have Infinity as the value in the retention policy, since we are doing this chain of multiplications: `24 * 60 * 60 * 1000`

Reply:
The max supported value is something along the lines of 169 years, or very close to it. I'll make sure it cannot be bigger than 100 years just to make sure